### DISCUSSION TITLE
Dynamically Rendered Child Livewire components throws multiple root detected warning

### DISCUSSION THREAD

**Discussion by ridxery** [livewire/livewire#3994] [7. Bugs (archived)]
Hi, I search through bugs and google but couldnt find anything.

Short info about app: I built a small SPA that dynamically change pages by triggering/emitting events and renders the corresponding livewire component as child inside the parent-livewire-component (lets call it spa-master-page). 

Problem: It just shows multiple root detected warning on dev console (everytime I dynamically render a child component) however livewire still works there is no problem. 
Interesting thing is if I refresh the page and livewire loads the same component (from 'query string' parameters) dynamically it doesnt show multiple root elements found warning on dev console, however when I emit my event (e.g. emit('changePage', xxx)) it does load the same component but throws multiple root detected warning however the same component that loaded on page refresh does not throw any warning (even though warning has been thrown livewire keeps working nothing broken but I just want to know why?).

livewire version: 2.6.5
laravel: 8.61.0

@joshhanley I can share private link for you to login and try the system on live and see by yourself that even though warning thrown everything working (even 3-4th level deeper components working just fine). its just so annoying little thing.

**Comment by joshhanley** [comment]
@ridxery thanks for reporting, I think that is the same issue as reported in #3887.

I haven't had a chance to see if I can fix it yet, but it shouldn't impact the functioning of the app, as they are just console warnings.

When I get some time though I'll see if we can get it fixed.

Hope this helps!

**Reply by ridxery** [reply]
Thanks to you @joshhanley as always fast and friendly reply. Its enough to know that wont effect my app and I havent done something wrong. I will contribute to livewire once I find some free time.

**Reply by joshhanley** [reply]
No worries! 🙂

**Comment by Tiagospem** [comment]


**Reply by ridxery** [reply]
Can you elaborate the 'UI' a bit? Also It would be helpful if you provide the php version - laravel version - livewire version.

**Reply by joshhanley** [reply]
@Tiagospem yeah can you please explain what you mean by a UI for form elements and exactly what breaks, along with screenshots and any relevant code?

Thanks!
